core/src/main/java/com/dtolabs/rundeck/core/schedule/JobScheduleManager.java

```diff
@@ -108,4 +111,24 @@
     default boolean scheduleRemoteJob(Map data){
         return false;
     }
+
+
+    /**
+     * Pre-run hook before a Job/adhoc is executed
+     *
+     * @param execution execution detail
+     * @return false to proceed, true to not proceed
```

Comment:
primarily because I thought the default interface method implementation should return false and have that mean no special behavior.  It could be opposite, i had that originally. I kinda dislike using a boolean actually.

Reply:
I like the enum idea.

Reply:
I read this a while back, i think it makes a lot of sense https://medium.com/better-programming/dont-use-boolean-arguments-use-enums-c7cd7ab1876a